tools/clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -1047,8 +1047,15 @@ bool CGDebugInfo::TryCollectHLSLRecordElements(const RecordType *Ty,
     unsigned VecSize = hlsl::GetHLSLVecSize(QualTy);
     unsigned ElemSizeInBits = CGM.getContext().getTypeSize(ElemQualTy);
     unsigned CurrentAlignedOffset = 0;
+    std::string FieldNameBuf;
     for (unsigned ElemIdx = 0; ElemIdx < VecSize; ++ElemIdx) {
-      StringRef FieldName = StringRef(&"xyzw"[ElemIdx], 1);
+      StringRef FieldName;
+      if (VecSize <= 4) {
+        FieldName = StringRef(&"xyzw"[ElemIdx], 1);
+      } else {
+        FieldNameBuf = "c" + std::to_string(ElemIdx);
```

Comment:
Is this "c" prefix naming convention used elsewhere?

@jeffnn - will this end up going directly through to PIX?

Reply:
Schweet

Reply:
This thread is still open. Is `c` an acceptable prefix for those fields?

Reply:
Nobody else seems to have an opinion.  We can always change if we get feedback on it while people are debugging shaders with it.

I assume the "c" stands for "component"?

Reply:
c for component makes sense to me. I asked Dwayne, who has extensive experience in ML scenarios, also and he's OK with it.